aana/core/models/speaker.py

```diff
@@ -0,0 +1,75 @@
+from typing import Annotated
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from aana.core.models.time import TimeInterval
+
+__all__ = [
+    "SpeakerDiarizationParams",
+    "SpeakerDiarizationSegment",
+    "SpeakerDiarizationSegments",
+]
+
+
+class SpeakerDiarizationParams(BaseModel):
+    """A model for the Speaker Diarization model parameters.
+
+    Attributes:
+        min_speakers (int): The minimum number of speakers present in the audio.
+        max_speakers (int): The maximum number of speakers present in the audio.
+    """
+
+    min_speakers: int = Field(
+        default=None,
+        description="The minimum number of speakers present in the audio.",
+    )
+
+    max_speakers: int = Field(
+        default=None,
+        description="The maximum number of speakers present in the audio.",
+    )
+
+    model_config = ConfigDict(
+        json_schema_extra={
+            "description": "Parameters for the spekaer diarization model.",
+        }
+    )
+
+
+class SpeakerDiarizationSegment(BaseModel):
+    """Pydantic schema for Segment from Speaker Diarization model.
+
+    Attributes:
+        time_interval (TimeInterval): The start and end time of the segment
+        speaker (str): speaker assignment of the model in the format "SPEAKER_XX"
+    """
+
+    time_interval: TimeInterval = Field(description="Time interval of the segment")
+    speaker: str = Field(description="speaker assignment from the model")
+
+    def to_whisper_dict(self) -> dict:
```

Comment:
I don't see it's being used anywhere. Is it a preparation for batched whisper?

Reply:
It will be needed when using with ASR but better would be to make it a generic dict such as `to_dict()`.